File: app/src/main/java/ru/ulstu_team/ulstuschedule/ui/common/teacher/TeacherScheduleFragment.java
package ru.ulstu_team.ulstuschedule.ui.common.teacher;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.Snackbar;
import android.support.v4.widget.SwipeRefreshLayout;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

import ru.ulstu_team.ulstuschedule.R;
import ru.ulstu_team.ulstuschedule.data.model.Lesson;
import ru.ulstu_team.ulstuschedule.databinding.ScheduleBinding;
import ru.ulstu_team.ulstuschedule.ui.base.BaseFragment;
import ru.ulstu_team.ulstuschedule.ui.common.StickyListScheduleAdapter;

public class TeacherScheduleFragment extends BaseFragment implements TeacherScheduleMvpView {

    public static final String TAG = "TeacherScheduleFragment";

    @Inject
    StickyListScheduleAdapter mAdapter;
    @Inject
    TeacherSchedulePresenter mPresenter;
    private View mView;
    private ScheduleBinding b;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);
        getTeacherScheduleComponent().inject(this);
        mView = inflater.inflate(R.layout.schedule, container, false);

        b = ScheduleBinding.bind(mView);
        mPresenter.attachView(this);
        mPresenter.loadSchedule();

        b.srlRefresh.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
            @Override
            public void onRefresh() {
                mPresenter.loadSchedule();
                b.srlRefresh.setRefreshing(false);
            }
        });

        return mView;
    }

    @Override
    public void showSchedule(List<Lesson> lessons) {
        mAdapter.setLessons(lessons, true);

        b.slTeacherLessons.setAdapter(mAdapter);
    }

    @Override
    public void showEmptySchedule() {
        mAdapter.setLessons(new ArrayList<Lesson>(), true);
    }

    @Override
    public void showError() {
        Snackbar.make(mView, "Возниикла ошибка", Snackbar.LENGTH_LONG)
                .setAction("Повторить", new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        reload();
                    }
                }).show();
    }

    @Override
    public void reload() {
        mPresenter.reload();
    }
}

File: app/src/main/java/ru/ulstu_team/ulstuschedule/ui/common/teacher/TeacherScheduleFragment.kt
package ru.ulstu_team.ulstuschedule.ui.common.teacher

import android.os.Bundle
import android.support.design.widget.Snackbar
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.schedule.*
import ru.ulstu_team.ulstuschedule.R
import ru.ulstu_team.ulstuschedule.data.model.Lesson
import ru.ulstu_team.ulstuschedule.ui.base.BaseFragment
import ru.ulstu_team.ulstuschedule.ui.common.StickyListScheduleAdapter
import java.util.*
import javax.inject.Inject

class TeacherScheduleFragment() : BaseFragment(), TeacherScheduleMvpView {

    @Inject
    internal lateinit var mAdapter: StickyListScheduleAdapter
    @Inject
    internal lateinit var mPresenter: TeacherSchedulePresenter
    private var mTeacherId = -1

    constructor(teacherId: Int) : this() {
        mTeacherId = teacherId
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        super.onCreateView(inflater, container, savedInstanceState)
        getTeacherScheduleComponent().inject(this)
        return inflater.inflate(R.layout.schedule, container, false)
    }

    override fun onStart() {
        super.onStart()
        mPresenter.attachView(this)
        loadSchedule()

        srlRefresh.setOnRefreshListener { loadSchedule() }
    }

    private fun loadSchedule() {
        if (mTeacherId <= 0)
            mPresenter.loadScheduleForCurrentTeacher()
        else
            mPresenter.loadScheduleForTeacher(mTeacherId)
    }

    override fun showSchedule(lessons: List<Lesson>) {
        mAdapter.setLessons(lessons, true)
        slTeacherLessons.adapter = mAdapter
    }

    override fun showEmptySchedule() {
        mAdapter.setLessons(ArrayList<Lesson>(), true)
    }

    override fun showError() {
        Snackbar.make(view, "Возниикла ошибка", Snackbar.LENGTH_LONG)
                .setAction("Повторить", { loadSchedule() })
                .show()
    }

    companion object {
        val TAG = "TeacherScheduleFragment"
    }
}
